src/realm/table.cpp

```diff
@@ -2536,8 +2536,68 @@ void Table::update_from_parent(size_t old_baseline) noexcept
     m_alloc.bump_storage_version();
 }
 
+void Table::schema_to_json(std::ostream& out, const std::map<std::string, std::string>& renames) const
+{
+    out << "{";
+    auto name = get_name();
+    if (renames.count(name))
+        name = renames.at(name);
+    out << "\"name\":\"" << name << "\"";
+    if (this->m_primary_key_col) {
+        out << ",";
+        out << "\"primaryKey\":\"" << this->get_column_name(m_primary_key_col) << "\"";
+    }
+    if (is_embedded()) {
+        out << ",\"isEmbedded\":true";
+    }
+    out << ",\"properties\":[";
+    auto col_keys = get_column_keys();
+    int sz = int(col_keys.size());
+    for (int i = 0; i < sz; ++i) {
+        auto col_key = col_keys[i];
+        name = get_column_name(col_key);
+        auto type = col_key.get_type();
+        if (renames.count(name))
+            name = renames.at(name);
+        out << "{";
+        out << "\"name\":\"" << name << "\"";
+        if (this->is_link_type(type)) {
+            out << ",\"type\":\"object\"";
+            name = this->get_opposite_table(col_key)->get_name();
+            if (renames.count(name))
+                name = renames.at(name);
+            out << ",\"object_type\":\"" << name << "\"";
+        }
+        else {
+            out << ",\"type\":\"" << get_data_type_name(DataType(type)) << "\"";
+        }
+        if (col_key.is_list()) {
+            out << ",\"isArray\":true";
+        }
+        if (col_key.is_set()) {
```

Comment:
To be clear, a Set of strings, would output as { name: "setField", type: "string", isArray: true, isSet: true }? Not an issue but just trying to understand

Reply:
A property is either a Set or a List. Adding "else if".